v1/remote/image_test.go

```diff
@@ -0,0 +1,109 @@
+package remote
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/authn"
+	"github.com/google/go-containerregistry/name"
+	"github.com/google/go-containerregistry/v1"
+	"github.com/google/go-containerregistry/v1/v1util"
+)
+
+type mockRegistry struct {
+	requests []*http.Request
+}
+
+// TODO(jonjohnsonjr): Replace this with better test data.
+const (
+	goodManifest = `{
+  "schemaVersion": 2,
+  "config": {
+    "digest": "sha256:deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef"
+  }
+}`
+
+	goodConfig = `{
+  "architecture": "amd64"
+}`
+)
+
+func TestManifestAndConfig(t *testing.T) {
```

Comment:
Follow some of the stuff I did for `remote.Write` (as discussed offline)

Reply:
Done - let me know if there's anything obvious I missed.

I implemented some digest checking, which added a bunch of tests as well.